pkg/controllers/machine/termination/controller.go

```diff
@@ -77,8 +78,8 @@ func (c *Controller) Finalize(ctx context.Context, machine *v1alpha5.Machine) (r
 		return reconcile.Result{}, err
 	}
 	for _, node := range nodes {
-		// We delete nodes to trigger the node finalization and deletion flow
-		if err = c.kubeClient.Delete(ctx, node); client.IgnoreNotFound(err) != nil {
+		// Trigger finalization to gracefully terminate the node, propagating the deletion grace period
+		if err = c.kubeClient.Delete(ctx, node, client.GracePeriodSeconds(lo.FromPtr(machine.DeletionGracePeriodSeconds))); client.IgnoreNotFound(err) != nil {
```

Comment:
As implemented, Karpenter will only ever terminate machines by setting `client.GracePeriodSeconds(math.MaxInt64)`. This will cascade down to Nodes.

In the manual use case, if a customer executes `kubectl delete machine`, that's intended as a force delete. In the future, we can consider making this configurable, similar to how pods have a configurable grace period.

Reply:
> In the manual use case, if a customer executes kubectl delete machine, that's intended as a force delete. 

Prior to this change, the "force delete" is meant in respect to `do-not-evict` pods and any other deprovisioning blocking conditions. I think we should make Karpenter's logic infer the same GracePeriodSeconds as the kubectl CLI would, which defaults to a non-zero DeletionGracePeriodSeconds. Making it a default --grace-period-seconds=0 sounds like a weird sharp edge that users have to be aware of any time they want to delete nodes/machines, as I know some users use the `kubectl delete machine` as a way to pick a Machine to be deleted, but not necessarily nuke the machine itself ASAP.